tests/typeparams/conversion_test.go

```diff
@@ -0,0 +1,250 @@
+package typeparams_test
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"runtime"
+	"testing"
+
+	"github.com/gopherjs/gopherjs/js"
+)
+
+// checkConversion is a general type conversion result checker.
+//
+// The expectation is that got and want have the same underlying Go type, and
+// they contain equal values. Src is the original value before type conversion,
+// provided for error message purposes.
+//
+// Note that this function is suitable for checking most conversion results
+// except conversion to interfaces. This is because use of reflect APIs requires
+// conversion to `any` interface, which must be assumed correct for this test to
+// be meaningful.
+func checkConversion(t *testing.T, src, got, want any) {
+	if reflect.TypeOf(got) != reflect.TypeOf(want) {
```

Comment:
should this have a `t.Helper()` call?

Reply:
Done. At first I omitted it because I felt seeing which line in checkConversion() triggered the error would be more useful, but today I feel the other way around ¯\\\_(ツ)\_/¯